### DISCUSSION TITLE
How to make Configuration persistent over ssh?

### DISCUSSION THREAD

**Discussion by defekT1312** [starship/starship#5766] [Q&A]
This is my prompt:
![grafik](https://github.com/starship/starship/assets/88479985/8fe95a85-1552-4216-ae1f-5afd032467a5)

And it falls back to this when I'm connected to an ssh host in my network:
![grafik](https://github.com/starship/starship/assets/88479985/c1d14132-39ea-431a-ba16-9e0ec8178b83)

This is my config:
```
[username]
style_user = 'green'
style_root = 'red'
format = '[$user]($style) '
disabled = false
show_always = true

# Konfiguriere die Ausgabe für den Text
[cmd_duration]
style = "neon_lilac"  # Neonlilafarbener Text
```

My Terminal is Guake and I'm using bash.
How can I make it on every System the same ?

**Comment by scrooop** [comment]
I'm not an expert but have looked into this same thing in the past. I think you need to install starship on the remote system and initialize it in the shell configuration file (.bashrc or .zshrc). I was bummed because I don't have administrator privileges on the remote server that I typically use so I'm pretty much out of luck.

**Reply by mrichar1** [reply]
You can install starship as an end-user without admin privileges by downloading the install script and then pecifying where to install - e.g.:

```
mkdir -p ~/.local/bin
sh install.sh --bin-dir=~/.local/bin
```

Then edit your .bashrc and make sure you have the following (changing `bash` as appropriate):

```
export PATH="~/.local/bin/:$PATH
eval "$(starship init bash)"
```

**Comment by scrooop** [comment]
@defekT1312 hey again — I recently started giving [Warp](https://www.warp.dev/) a try, and I just realized they have [a built-in method](https://docs.warp.dev/features/ssh) for maintaining  not only your prompt style but also a ton of other really useful features that wouldn't normally carry over when you connect to a remote server that you haven't/can't configure. And they play well with starship too. I was pretty die hard iTerm but I'm really loving Warp... You might check it out.

**Comment by esseti** [comment]
but then you have to copy the file on every remote, or am I missing something? (i'm not referring to warp, but a general terminal case)
